modules/nf-core/star/genomegenerate/main.nf

```diff
@@ -4,12 +4,13 @@ process STAR_GENOMEGENERATE {
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/mulled-v2-1fa26d1ce03c295fe2fdcf85831a92fbcbd7e8c2:ded3841da0194af2701c780e9b3d653a85d27489-0' :
-        'biocontainers/mulled-v2-1fa26d1ce03c295fe2fdcf85831a92fbcbd7e8c2:ded3841da0194af2701c780e9b3d653a85d27489-0' }"
+        'https://depot.galaxyproject.org/singularity/star:2.7.11b--h43eeafb_1' :
+        'biocontainers/star:2.7.11b--h43eeafb_1' }"
 
     input:
-    tuple val(meta), path(fasta)
-    tuple val(meta2), path(gtf)
+    tuple val(meta) , path(fasta)
+    tuple val(meta2), path(fai)
```

Comment:
I'm following the example of a bunch of other modules, AFAIK the guidelines aren't adamant on this. (Can't find where it's specified though)

Reply:
I didn't mean to suggest that your approach is against the rules. 

To me, it just seemed easier to keep the Fasta and its associated index in one channel rather than in two separate (which might get out of sync and become disordered). 

But if other modules handle it with separate channels, I am fine with that, too. It does not require some possibly confusing `.map()` operation to pass it to the tool.

Reply:
Oh yeah, I agree about that. TBH, when this gets merged, the first thing I do is patch the module and make one tuple of all the args, but that's just for me

Reply:
Until we have a proper way to deal with it in an automatic way, I'd rather we keep each single reference file separated, otherwise it might get complex and messy.
But I agree that I don't like this too much

Reply:
To me, an index is just the _table of contents_ for the reference, so conceptually not a separate entity. I think, there is simply little use for it on its own.

Personally, I find it convenient, if it is automatically attached to the source file like [42basepairs](https://42basepairs.com) does, but I see why this opinionated approach is discouraged for a module. Thus, I think you were correct separating the channels.

Reply:
I think @maxulysse is right here. We should probably wait to change this before this can be properly implemented